tests/test_cms_config.py

```diff
@@ -17,6 +24,96 @@
 from djangocms_versioning.test_utils.polls.models import Poll, PollContent
 
 
+req_factory = RequestFactory()
+
+
+class PageContentVersioningBehaviourTestCase(CMSTestCase):
+
+    def setUp(self):
+        self.site = Site.objects.get_current()
+        self.user = self.get_superuser()
+        self.language = 'en'
+        self.title = 'test page'
+
+        self.page = create_page(
+            title=self.title,
+            language=self.language,
+            template='page.html',
+            menu_title='test page',
+            in_navigation=True,
+            limit_visibility_in_menu=None,
+            site=self.site,
+            created_by=self.user,
+        )
+        self.content = PageContent._original_manager.get(page=self.page.pk)
+        self.version = self.content.versions.all()[0]
+
+    def test_saving_draft_has_pageurl(self):
+        '''A draft has pageurl as normal'''
+        url = self.page.get_urls().first()
+
+        self.assertEqual(self.version.state, DRAFT)
+        self.assertEqual(url.path, slugify(self.title))
+
+    def test_publishing_draft_retains_pageurl(self):
+        '''We publish a version, pageurl is still the same'''
+        self.version.publish(self.user)
+
+        url = self.page.get_urls().first()
+        self.assertEqual(url.path, slugify(self.title))
+
+    def test_published_version_with_new_version_retains_pageurl(self):
+        '''We have a published, we create a new version and publish that.
+        PageUrl path stays the same'''
+        self.version.publish(self.user)
+        v2 = self.version.copy(self.user)
+        v2.publish(self.user)
+
+        page = Page.objects.get(pk=self.page.pk)
+        url = page.get_urls().first()
+
+        self.assertEqual(url.path, slugify(self.title))
+
+    def test_published_version_with_new_version_retains_pageurl_unmanaged(self):
+        '''We have a published, we create a new version and publish that.
+        The url in question is unmanaged and stays the same.'''
+        url = self.page.get_urls().first()
+        url.managed = False
+        url.save()
+
+        self.version.publish(self.user)
+        v2 = self.version.copy(self.user)
+        v2.publish(self.user)
+
+        page = Page.objects.get(pk=self.page.pk)
+        url = page.get_urls().first()
+
+        self.assertEqual(url.path, slugify(self.title))
```

Comment:
This test is the one that will fail if we change add https://github.com/divio/djangocms-versioning/pull/203/files#diff-8b290a1c3759be311fb10a9b1a4faa88L216